File: Medium/2026-05-01-396-Rotate-Function/solution.js
var maxRotateFunction = function(nums) {
    const n = nums.length;
    let F = 0;
    let totalSum = 0;

    for (let i = 0; i < n; i++) {
        F += i * nums[i];
        totalSum += nums[i];
    }

    let maxF = F;

    for (let k = 1; k < n; k++) {
        F = F + totalSum - n * nums[n - k];
        maxF = Math.max(maxF, F);
    }

    return maxF;
};

File: Medium/2026-05-01-396-Rotate-Function/solution.py
class Solution:
    def maxRotateFunction(self, nums: List[int]) -> int:
        n = len(nums)
        total_sum = sum(nums)
        current_value = sum(i * num for i, num in enumerate(nums))
        max_value = current_value
        
        for k in range(1, n):
            current_value += total_sum - n * nums[n - k]
            max_value = max(max_value, current_value)
        
        return max_value